jolie/src/main/java/jolie/Interpreter.java

```diff
@@ -1004,7 +1014,9 @@ private void init( State initState )
 		} else {
 			sessionStarters = Collections.unmodifiableMap( sessionStarters );
 			try {
-				initExecutionThread = new InitSessionThread( this, getDefinition( "init" ), initState );
+				InitDefinitionProcess initProcess = (InitDefinitionProcess) getDefinition( "init" );
+				var future = initProcess.getFuture();
+				initExecutionThread = new InitSessionThread( this, initProcess, initState );
 
```

Comment:
Why don't you just make the future a parameter of the constructor of `InitSessionThread`? It should give a much simpler implementation.